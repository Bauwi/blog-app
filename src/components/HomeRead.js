import React, { Component } from 'react';
import { connect } from 'react-redux';

import Header from './Header';
import PublicPostsList from './PublicPostsList';

import { startSetPostsSample } from '../actions/posts';
import selectPosts from '../selectors/posts';

export class HomeRead extends Component {
  componentWillMount() {
    this.props.startSetPostsSample(20);
  }

  render() {
    console.log(this.props);
    return (
      <div>
        <Header />HomeRead
        <PublicPostsList posts={this.props.posts} />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startSetPostsSample: sampleSize => dispatch(startSetPostsSample(sampleSize))
});

const mapStateToProps = state => ({
  posts: selectPosts(state.posts, state.filters)
});

export default connect(mapStateToProps, mapDispatchToProps)(HomeRead);
