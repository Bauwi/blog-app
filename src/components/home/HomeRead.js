import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import HeaderSub from '../header/HeaderSub';
import Jumbotron from '../header/Jumbotron';
import PublicPostsList from '../PublicPostsList';
import LoadingPage from '../LoadingPage';
import PopularLasttFilter from '../filters/PopularLastFilter';

import { startSetPostsSample } from '../../actions/readings';

export class HomeRead extends Component {
  componentDidMount() {
    this.props.startSetPostsSample();
  }

  renderCustomCategories() {
    return this.props.categories.map((category, i) => {
      const grid = i === 0 ? 'grid-home-second' : 'grid-home-first';
      const range = i === 0 ? 8 : i === 1 ? 8 : i === 2 ? 8 : 0;
      return <PublicPostsList key={category} grid={grid} category={category} range={range} />;
    });
  }

  render() {
    return (
      <div className="page-container">
        <HeaderSub />
        <Jumbotron />

        {this.props.isLoading ? (
          <LoadingPage />
        ) : (
          <div>
            <div className="popular-last-filterbar__container">
              <div className="content-container">
                <PopularLasttFilter relative />
              </div>
            </div>
            <PublicPostsList grid="grid-home-first" category="all" range={17} />
          </div>
        )}
        {!this.props.isLoading && this.renderCustomCategories()}
      </div>
    );
  }
}

HomeRead.propTypes = {
  categories: PropTypes.array.isRequired,
  startSetPostsSample: PropTypes.func.isRequired
};

const mapDispatchToProps = dispatch => ({
  startSetPostsSample: () => dispatch(startSetPostsSample())
});

const mapStateToProps = state => ({
  categories: state.users.preferences
    ? state.users.preferences.topCategories
    : ['architecture', 'cinema', 'music'],
  isLoading: state.readings.isLoading
});

export default connect(mapStateToProps, mapDispatchToProps)(HomeRead);
