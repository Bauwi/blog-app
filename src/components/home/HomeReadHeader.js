import React from 'react';
import PopularLasttFilter from '../filters/PopularLastFilter';

export const HomeReadHeader = () => (
  <div className="content-container">
    <PopularLasttFilter />
  </div>
);

export default HomeReadHeader;
