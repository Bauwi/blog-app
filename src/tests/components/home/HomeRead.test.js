import React from 'react';
import { shallow } from 'enzyme';
import { HomeRead } from '../../../components/home/HomeRead';

test('should render HomeRead component properly', () => {
  const wrapper = shallow(<HomeRead categories={[]} startSetPostsSample={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

test('should have rendered custom categories properly', () => {
  const wrapper = shallow(<HomeRead categories={['alpha', 'beta', 'gamma']} startSetPostsSample={() => {}} />);

  expect(wrapper).toMatchSnapshot();
});
