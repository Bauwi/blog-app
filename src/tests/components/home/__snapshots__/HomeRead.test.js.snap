// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have rendered custom categories properly 1`] = `
<div
  className="page-container"
>
  <HeaderSub />
  <withRouter(Connect(Jumbotron)) />
  <div>
    <div
      className="popular-last-filterbar__container"
    >
      <div
        className="content-container"
      >
        <Connect(PopularLasttFilter)
          relative={true}
        />
      </div>
    </div>
    <Connect(PublicPostsList)
      category="all"
      grid="grid-home-first"
      range={14}
    />
  </div>
  <Connect(PublicPostsList)
    category="alpha"
    grid="grid-home-second"
    key="alpha"
    range={7}
  />
  <Connect(PublicPostsList)
    category="beta"
    grid="grid-home-first"
    key="beta"
    range={6}
  />
  <Connect(PublicPostsList)
    category="gamma"
    grid="grid-home-first"
    key="gamma"
    range={3}
  />
</div>
`;

exports[`should render HomeRead component properly 1`] = `
<div
  className="page-container"
>
  <HeaderSub />
  <withRouter(Connect(Jumbotron)) />
  <div>
    <div
      className="popular-last-filterbar__container"
    >
      <div
        className="content-container"
      >
        <Connect(PopularLasttFilter)
          relative={true}
        />
      </div>
    </div>
    <Connect(PublicPostsList)
      category="all"
      grid="grid-home-first"
      range={14}
    />
  </div>
</div>
`;
